inspire_crawler/tasks.py

```diff
@@ -200,6 +203,18 @@ def schedule_crawl(spider, workflow, **kwargs):
     """Schedule a crawl using configuration from the workflow objects."""
     from inspire_crawler.utils import get_crawler_instance
 
+    queue_size_limit = kwargs.get('queue_size_limit')
+    if queue_size_limit:
+        default_queue =  current_app.config.get('CRAWLER_CELERY_QUEUE', 'celery')
+        spider_results_queue_name = current_app.config['CELERY_QUEUE_SPIDER_MAPPING'].get(spider, default_queue)
+        with current_celery_app.connection_or_acquire() as conn:
+            current_queue_size = conn.default_channel.queue_declare(
+                queue=spider_results_queue_name
+            )
+        if current_queue_size > queue_size_limit:
+            current_app.logger.info('Queue is full. Current size: {}. Skipping crawl'.format(current_queue_size))
+            return
+
```

Comment:
If we integrate it this way, aren't we going to lose results? Somehow we will need to put them back to the queue. Am I missing something?

Reply:
we meant it for desy, and in case of desy harvest, in desy harvest what we do is:
* We have a crawl scheduled every hour (so each hour we send `schedule_crawl` (the spider hasn't run yet)
* During `schedule_crawl` we move jsonl to the processed bucket
* Since we won't run the spider if the queue is full, jsonl is still in the `processed` bucket, so it will be in `incoming` bucket and will be crawled in next `schedule_crawl`

Maybe I am wrong, but I think it's gonna work like that.

Reply:
and the `queue_size_limit` comes from the celerybeat config right?

Reply:
yes, that was the idea